cloud/kyup/kyup.py

```diff
@@ -0,0 +1,338 @@
+    kyup: >
+      api_key=XXXX
+      action=restart
+      name=avalon
+
+'''
+
+import os
+import requests
+import time
+import base64
+import md5
+from Crypto.Cipher import AES
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+# import module snippets
+from ansible.module_utils.basic import *
+
+## Defaults for the module
+# API URL
+url='https://api.kyup.com/client/v1'
+
+# Modify this counter if you want to wait for the requests, more then a minute
+retries=12 # 12 x 5sec = 60 seconds
+
+
+## Code used for encryption of the passwords
+blocksize = AES.block_size
+
+def pad( data ):
+    padding = blocksize - ( len(data) % blocksize )
+    return data + (chr(padding) * padding)
+
+def encrypt( key, data ):
+    data = pad(data)
+    m = md5.new(key)
+    iv = m.hexdigest()[0:blocksize]
+    cipher = AES.new( key, AES.MODE_CBC, iv )
+    return base64.b64encode( cipher.encrypt( data ) )
+## End of encryption code
+
+def api_request(module, data):
+    ret = {}
+    count = retries;
+    while count != 0:
+        # get the response
+        resp = requests.post(url, { 'request': data })
+        # parse the json
+        ret = json.loads(resp.text)
+        if ret['status']:
+            return ret
+        count -= 1
+        time.sleep(5)
+
+    if 'status' in ret:
+        if ret['status']:
+            return ret
+        else:
+            if 'data' in ret and 'error_code' in ret['data']:
+                if ret['data']['error_code'] == 101:
+                    module.fail_json(changed=False, msg = 'Failed to execute request. Error code: %d Error msg: %s req: %s' % (ret['data']['error_code'], ret['data']['error'], data))
+                else:
+                    module.fail_json(changed=False, msg = 'Failed to execute request. Error code: %d Error msg: %s' % (ret['data']['error_code'], ret['data']['error']))
+    else:
+        module.fail_json(changed=False, msg = 'Req: ' + data + ' Resp: ' + ret)
+
+def kyup_action(module, action, container_id = 0):
+    req = '{"action":"cloudList","authorization_key":"%s","data":{}}'
+    ret = api_request(module, req % module.params.get('api_key'))
+    for container in ret['data']['list']:
+        if container['name'] == module.params.get('name'):
+          container_id = container['id']
+          break
+
+    if container_id == 0:
+        module.exit_json(changed=False, msg = 'Could not find container id for container name %s' % module.params.get('name'))
+
+    req = '{"action":"%s","authorization_key":"%s","data":{"container_id":%d}}'
+    ret = api_request(module, req % (action, module.params.get('api_key'), int(container_id)) )
+    if action == 'cloudDetails':
+        return ret['data']['container']
+    else:
+        module.exit_json(changed=True, msg = '%s successfull' % action)
+
+def get_task_status(module, task_id):
+    req = '{"action":"getTask","authorization_key":"%s","data":{"task_id":%d}}'
+    count = retries;
+    while count != 0:
```

Comment:
I do prefer to actually write it this way... but if you require it like that. I'll change it.